examples/sparse_blas/compile_time_dispatching/sparse_blas_spmv_usm_mklcpu_cusparse.cpp

```diff
@@ -257,17 +246,40 @@ void print_example_banner() {
 int main(int /*argc*/, char ** /*argv*/) {
     print_example_banner();
 
+    auto exception_handler = [](sycl::exception_list exceptions) {
+        for (std::exception_ptr const &e : exceptions) {
+            try {
+                std::rethrow_exception(e);
+            }
+            catch (sycl::exception const &e) {
+                std::cout << "Caught asynchronous SYCL "
+                             "exception during sparse::spmv:\n"
+                          << e.what() << std::endl;
+            }
+        }
+    };
+
     try {
-        // TODO: Add cuSPARSE compile-time dispatcher in this example once it is supported.
-        sycl::device cpu_dev(sycl::cpu_selector_v);
+        sycl::queue cpu_queue(sycl::cpu_selector_v, exception_handler);
+        sycl::queue gpu_queue(sycl::gpu_selector_v, exception_handler);
+        unsigned int vendor_id = gpu_queue.get_device().get_info<sycl::info::device::vendor_id>();
+        if (vendor_id != NVIDIA_ID) {
+            std::cerr << "FAILED: NVIDIA GPU device not found" << std::endl;
+            return 1;
+        }
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::mklcpu> cpu_selector{ cpu_queue };
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cusparse> gpu_selector{ gpu_queue };
 
-        std::cout << "Running Sparse BLAS SPMV USM example on CPU device." << std::endl;
-        std::cout << "Device name is: " << cpu_dev.get_info<sycl::info::device::name>()
+        std::cout << "Running Sparse BLAS SPMV USM example on:" << std::endl;
+        std::cout << "\tCPU device: " << cpu_queue.get_device().get_info<sycl::info::device::name>()
+                  << std::endl;
+        std::cout << "\tGPU device: " << gpu_queue.get_device().get_info<sycl::info::device::name>()
                   << std::endl;
         std::cout << "Running with single precision real data type:" << std::endl;
 
-        run_sparse_matrix_vector_multiply_example<float, std::int32_t>(cpu_dev);
-        std::cout << "Sparse BLAS SPMV USM example ran OK." << std::endl;
+        run_sparse_matrix_vector_multiply_example<float, std::int32_t>(cpu_selector);
+        run_sparse_matrix_vector_multiply_example<float, std::int32_t>(gpu_selector);
```

Comment:
What happens to this example if the library is built only with cuSPARSE but not with mklcpu backend? Wouldn't it lead to build errors (since this appears to be a compile-time-dispatching example)? Maybe this belongs in the runtime dispatching instead, or maybe this example needs further modification (maybe through `#ifdef` flags) so it compiles and runs for any of {mklcpu only, cusparse only, {mklcpu, cusparse}} backend combinations?

Reply:
The example is compiled only if both MKLCPU and NVIDIA backends are enabled. This is controlled by CMake here: https://github.com/oneapi-src/oneMKL/pull/527/files#diff-756306f0912d07033590d9f00855a0e97e14ed76343889a773744f7cd715036cR22
I've used the same pattern as the other domains.

Reply:
I think ideally we should allow independently compiling the example for CPU and GPU backends instead of only when both backends are built, but I suppose this is fine too.